run-tests.php

```diff
@@ -860,9 +860,22 @@ function write_information(): void
     }
     @unlink($info_file);
 
-    // load list of enabled extensions
-    save_text($info_file,
-        '<?php echo str_replace("Zend OPcache", "opcache", implode(",", get_loaded_extensions())); ?>');
+    // load list of enabled and loadable extensions
+    save_text($info_file, <<<'PHP'
+        <?php
+        echo str_replace("Zend OPcache", "opcache", implode(",", get_loaded_extensions()));
+        $ext_dir = ini_get("extension_dir");
+        foreach (scandir($ext_dir) as $file) {
+            if (!preg_match('/^(?:php_)?([_a-zA-Z0-9]+)\.(?:so|dll)$/', $file, $matches)) {
+                continue;
+            }
+            $ext = $matches[1];
+            if (!extension_loaded($ext) && @dl($file)) {
```

Comment:
Is the dl() check here necessary for something (unrelated DLLs in the directory?) or just defensive?

Reply:
At least with this regex, unrelated DLLs would be caught (e.g. php8.dll). We could make the regex more strict, but I think actually trying to load the extension could prevent some issues (such as having an extension for another PHP version in extension_dir).